keras/src/models/model.py

```diff
@@ -156,6 +156,15 @@ def __init__(self, *args, **kwargs):
             functional.Functional.__init__(self, *args, **kwargs)
         else:
             Layer.__init__(self, *args, **kwargs)
+            if args:
+                inputs = args[0]
+            # Only set _input_names if not already initialized by Functional.
```

Comment:
You're in an else, you know for sure it's not a Functional model if you're here.